libnetwork/networkdb/delegate.go

```diff
@@ -148,8 +148,8 @@ func (nDB *NetworkDB) handleTableEvent(tEvent *TableEvent, isBulkSync bool) bool
 	// Update our local clock if the received messages has newer time.
 	nDB.tableClock.Witness(tEvent.LTime)
 
+	nDB.Lock()
```

Comment:
Now that we're no longer micro-managing locking, would it make sense to defer the unlock?

```go
nDB.Lock()
defer nDB.Unlock()
```

At a quick glance, only the last part of the function runs without a lock, and (again, at a glance) possibly won't do harm to run it while still under lock?
https://github.com/moby/moby/blob/def9078135556e3bc80e1838390850db22861a96/libnetwork/networkdb/delegate.go#L208-L235